src/creature.cpp

```diff
@@ -188,7 +188,10 @@ bool Creature::sees( const tripoint &t, bool is_player ) const
         }
         if( is_player ) {
             // Special case monster -> player visibility, forcing it to be symmetric with player vision.
-            return range >= wanted_range &&
+            const float player_visibility_factor = ( g->u.visibility() ) / 100.0f;
+            int adj_range = std::floor( static_cast<float>( range ) * player_visibility_factor );
```

Comment:
The explicit cast is not needed. The second operand is a `float`, so the first will implicitly casted to one as well.